core/lib/Cardano/Address/Style/Shelley.hs

```diff
@@ -537,163 +560,245 @@ inspectShelleyAddress
+    0b00010000 | addrRestLength == credentialHashSize + credentialHashSize ->
+        Right addressInfo
+            { infoStakeReference = Just ByValue
+            , infoScriptHash = Just addrHash1
+            , infoStakeKeyHash = Just addrHash2
+            }
+    -- 0010: base address: keyhash28,scripthash32
+    0b00100000 | addrRestLength == credentialHashSize + credentialHashSize ->
+        Right addressInfo
+            { infoStakeReference = Just ByValue
+            , infoSpendingKeyHash = Just addrHash1
+            , infoStakeScriptHash = Just addrHash2
+            }
+    -- 0011: base address: scripthash32,scripthash32
+    0b00110000 | addrRestLength == 2 * credentialHashSize ->
+        Right addressInfo
+            { infoStakeReference = Just ByValue
+            , infoScriptHash = Just addrHash1
+            , infoStakeScriptHash = Just addrHash2
+            }
+    -- 0100: pointer address: keyhash28, 3 variable length uint
+    0b01000000 | addrRestLength > credentialHashSize -> do
+        ptr <- getPtr addrHash2
+        pure addressInfo
+            { infoStakeReference = Just $ ByPointer ptr
+            , infoSpendingKeyHash = Just addrHash1
+            }
+    -- 0101: pointer address: scripthash32, 3 variable length uint
+    0b01010000 | addrRestLength > credentialHashSize -> do
+        ptr <- getPtr addrHash2
+        pure addressInfo
+            { infoStakeReference = Just $ ByPointer ptr
+            , infoScriptHash = Just addrHash1
+            }
+    -- 0110: enterprise address: keyhash28
+    0b01100000 | addrRestLength == credentialHashSize ->
+        Right addressInfo
+            { infoStakeReference = Nothing
+            , infoSpendingKeyHash = Just addrHash1
+            }
+    -- 0111: enterprise address: scripthash32
+    0b01110000 | addrRestLength == credentialHashSize ->
+        Right addressInfo
+            { infoStakeReference = Nothing
+            , infoScriptHash = Just addrHash1
+            }
+    -- 1110: reward account: keyhash28
+    0b11100000 | addrRestLength == credentialHashSize ->
+        Right addressInfo
+            { infoStakeReference = Just ByValue
+            , infoStakeKeyHash = Just addrHash1
+            }
+    -- 1111: reward account: scripthash32
+    0b11110000 | addrRestLength == credentialHashSize ->
+        Right addressInfo
+            { infoStakeReference = Just ByValue
+            , infoScriptHash = Just addrHash1
+            }
+    unknown -> Left (UnknownType unknown)
 
-    getPtr :: (Alternative m, MonadThrow m) => ByteString -> m ChainPointer
+  where
+
+    addressInfo = AddressInfo
+        { infoNetworkTag = NetworkTag $ fromIntegral addrNetwork
+        , infoStakeReference = Nothing
+        , infoSpendingKeyHash = Nothing
+        , infoStakeKeyHash = Nothing
+        , infoScriptHash = Nothing
+        , infoStakeScriptHash = Nothing
+        }
+
+    -- TODO: Could do something better for pointer and try decoding the
+    -- pointer.
```

Comment:
I think this comment is now outdated because the function below does decode the pointer.